Modules/_json.c

```diff
@@ -1268,38 +1268,39 @@ encoder_call(PyEncoderObject *self, PyObject *args, PyObject *kwds)
 {
     /* Python callable interface to encode_listencode_obj */
     static char *kwlist[] = {"obj", "_current_indent_level", NULL};
-    PyObject *obj, *result;
+    PyObject *obj;
     Py_ssize_t indent_level;
-    _PyUnicodeWriter writer;
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "On:_iterencode", kwlist,
-        &obj, &indent_level))
+                                     &obj, &indent_level))
         return NULL;
 
-    _PyUnicodeWriter_Init(&writer);
-    writer.overallocate = 1;
+    PyUnicodeWriter *writer = PyUnicodeWriter_Create(0);
+    if (writer == NULL) {
+        return NULL;
+    }
 
     PyObject *newline_indent = NULL;
     if (self->indent != Py_None) {
         newline_indent = _create_newline_indent(self->indent, indent_level);
         if (newline_indent == NULL) {
-            _PyUnicodeWriter_Dealloc(&writer);
+            PyUnicodeWriter_Discard(writer);
             return NULL;
         }
     }
-    if (encoder_listencode_obj(self, &writer, obj, newline_indent)) {
-        _PyUnicodeWriter_Dealloc(&writer);
+    if (encoder_listencode_obj(self, writer, obj, newline_indent)) {
+        PyUnicodeWriter_Discard(writer);
         Py_XDECREF(newline_indent);
         return NULL;
     }
     Py_XDECREF(newline_indent);
 
-    result = PyTuple_New(1);
-    if (result == NULL ||
-            PyTuple_SetItem(result, 0, _PyUnicodeWriter_Finish(&writer)) < 0) {
-        Py_XDECREF(result);
+    PyObject *str = PyUnicodeWriter_Finish(writer);
+    if (str == NULL) {
         return NULL;
     }
+    PyObject *result = PyTuple_Pack(1, str);
```

Comment:
I think this new code is cleaner than the old code. But `PyTuple_Pack` is a bit slow (see https://github.com/python/cpython/issues/118222), so we should measure whether this change makes an impact

Reply:
See my comment below. It's not faster, it's not slower. The change has no significant effect on performance.

Reply:
I don't think that PyTuple_Pack() is the performance bottleneck of this function.

I chose to use PyTuple_Pack() to avoid creating an incomplete tuple and then fill it. I prefer an atomic API which creates directly valid tuple.